### PR TITLE
Added Guard to non fungible chain accounts

### PR DESCRIPTION
**Pull request by MRVDH**
image

### CODE REVIEW

File: packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts [hunk 1]
```diff
@@ -27,21 +30,43 @@ export default builder.node(
...
+        complexity: COMPLEXITY.FIELD.CHAINWEB_NODE,
+        async resolve(parent) {
+          try {
+            const tokenDetails = await tokenDetailsLoader.load({
+              accountName: parent.accountName,
+              chainId: parent.chainId,
+            });
+
+            const accountDetails = await nonFungibleAccountDetailsLoader.load({
+              tokenId: tokenDetails[0].id,
```
Comment (MRVDH) [inline packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts:55]: Can we make the assumption that for this account and chain the guard is the same for all tokens? @alber70g

Comment (alber70g) [reply packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts:55]: No we cannot :)
Basically each token has it's own tables. So also it's own accountnames, guards and balances. So they can have a different guard as well

**Review by sstraatemans — APPROVED** [review]

File: packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  balance: number;
+  guard: {
+    keys: string[];
+    pred: Guard['predicate'];
+  };
+};
+
+function getClient(chainId: string, networkId: string): IClient {
+  return createClient(
+    `${dotenv.NETWORK_HOST}/chainweb/0.0/${networkId}/chain/${chainId}/pact`,
```
Comment (nil-amrutlal-dept) [inline packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts:22]: There's an `apiVersion` property on network that can be used to replace the hardcoded `0.0`.

File: packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  );
+}
+
+export async function getNonFungibleAccountDetails(
+  tokenId: string,
+  accountName: string,
+  chainId: string,
+): Promise<NonFungibleChainAccountDetails | null> {
+  let result;
+  const networkId = (await networkConfig).networkId;
```
Comment (nil-amrutlal-dept) [inline packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts:32]: Regarding the `apiVersion`

```suggestion
  const { networkId, apiVersion } = await networkConfig;
```

**Review by nil-amrutlal-dept — APPROVED** [review]

File: packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts (hunk 2 above)
Comment (MRVDH) [reply packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts:22]: Feedback applied 👍

File: packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts (hunk 3 above)
Comment (MRVDH) [reply packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts:32]: Feedback applied 👍

File: packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts (hunk 1 above)
Comment (MRVDH) [reply packages/apps/graph/src/graph/objects/non-fungible-chain-account.ts]: Feedback applied 👍

**Pull request merged**